arcgis-ios-sdk-samples.xcodeproj/project.pbxproj

```diff
@@ -5049,6 +5073,26 @@
 /* End PBXResourcesBuildPhase section */
 
 /* Begin PBXShellScriptBuildPhase section */
+		007E8EF32578290F003077CC /* Create .secrets file if it does not exist */ = {
+			isa = PBXShellScriptBuildPhase;
+			alwaysOutOfDate = 1;
+			buildActionMask = 2147483647;
+			files = (
+			);
+			inputFileListPaths = (
+			);
+			inputPaths = (
+			);
+			name = "Create .secrets file if it does not exist";
+			outputFileListPaths = (
+			);
+			outputPaths = (
+				"$(SRCROOT)/.secrets",
```

Comment:
According to a post on SO: https://stackoverflow.com/a/63236362/14369688 I found adding this line helpful, instead of running the whole script in pre-action. A few notes

1. According to Phil, https://github.com/Esri/arcgis-runtime-samples-ios/pull/1024#issuecomment-745130935 adding stuff to Scheme's pre-action is not ideal.
2. The original problem is that the build system doesn't pick up the newly created `.secrets` file during a build. 
3. Later it turned out to be possible by specifying the output path in the Run Script phase. So that the build system knows that this file might change and will detect updates before running other phases that depend on it (including the build rule)